src/features/surveys/store.ts

```diff
@@ -121,6 +121,31 @@ const surveysSlice = createSlice({
       item.isLoading = false;
       item.loaded = new Date().toISOString();
     },
+    surveySubmissionUpdate: (
+      state,
+      action: PayloadAction<[number, string[]]>
+    ) => {
+      const [submissionId, mutating] = action.payload;
+      const item = state.submissionList.items.find(
+        (item) => item.id == submissionId
+      );
+      if (item) {
+        item.mutating = mutating;
+      }
+    },
+    surveySubmissionUpdated: (
+      state,
+      action: PayloadAction<ZetkinSurveySubmission>
+    ) => {
+      const submission = action.payload;
+      const item = state.submissionList.items.find(
+        (item) => item.id == submission.id
+      );
+      if (item) {
+        item.data = { ...item.data, ...submission } as ZetkinSurveySubmission;
```

Comment:
Nice! Isn't `immer` great? 😄

Reply:
But why is the type assertion (`as`) necessary?

Reply:
I suppose it's not. Easy to get paranoid with the TypeScript typing system :sweat_smile: